apps/ios/Shared/Views/Chat/ChatInfoView.swift

```diff
@@ -701,12 +719,38 @@ private struct CallButton: View {
 
         InfoViewButton(image: image, title: title, disabledLook: !canCall, width: width) {
             if canCall {
-                if CallController.useCallKit() {
-                    CallController.shared.startCall(contact, mediaType)
-                } else {
-                    // When CallKit is not used, colorscheme will be changed and it will be visible if not hiding sheets first
-                    dismissAllSheets(animated: true) {
-                        CallController.shared.startCall(contact, mediaType)
+                if let connStats = connectionStats {
+                    if connStats.ratchetSyncState == .ok {
+                        if CallController.useCallKit() {
+                            CallController.shared.startCall(contact, mediaType)
+                        } else {
+                            // When CallKit is not used, colorscheme will be changed and it will be visible if not hiding sheets first
+                            dismissAllSheets(animated: true) {
+                                CallController.shared.startCall(contact, mediaType)
+                            }
+                        }
+                    } else if connStats.ratchetSyncAllowed {
+                        showAlert(SomeAlert(
+                            alert: Alert(
+                                title: Text("Fix connection?"),
+                                message: Text("Connection requires encryption renegotiation."),
+                                primaryButton: .default(Text("Fix")) {
+                                    Task {
+                                        await syncContactConnection(contact, $connectionStats, force: false, showAlert: showAlert)
```

Comment:
each call to syncContactConnection produces "Passing argument of non-sendable type..." warning